api/app/routers/hfi_calc.py

```diff
@@ -27,37 +30,94 @@
 )
 
 
+def load_request(request: HFIResultRequest) -> HFIResultRequest:
+    """ If we need to load the request from the database, we do so. Otherwise we return the request as is. """
+    if request.start_date is None:
+        with app.db.database.get_read_session_scope() as session:
+            stored_request = get_most_recent_updated_hfi_request(session, request.selected_fire_center_id)
+            if stored_request:
+                return HFIResultRequest.parse_obj(stored_request.request)
+    return request
+
+
+def save_request(request: HFIResultRequest, username: str):
+    """ Save the request to the database (if there's a valid prep period) """
+    if request.start_date is not None and request.end_date is not None:
+        with app.db.database.get_write_session_scope() as session:
+            store_hfi_request(session, request, username)
+    return request
+
+
+def validate_date_range(start_date: date, end_date: date):
+    """ Sets the start_date to today if it is None.
```

Comment:
is it necessary to also validate that the date range is a minimum of 2 days? I believe feedback we got from the last sprint review was that prep periods can be anywhere from 2-7 days inclusive.

Reply:
Oh dear - I missed that - I'll update.

Reply:
I'll default to 5, limit to 7 max and min 2.